### PR TITLE
Add modern-go libraries to unwanted dependencies

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Track additional problematic libraries to burn down usage of.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/cc @dims
/milestone v1.33

### CODE REVIEW

File: hack/unwanted-dependencies.json [hunk 1]
```diff
@@ -204,6 +206,23 @@
...
+        "go.etcd.io/etcd/client/v2",
+        "go.etcd.io/etcd/server/v3",
+        "go.opentelemetry.io/contrib/instrumentation/github.com/emicklei/go-restful/otelrestful",
+        "k8s.io/kube-openapi",
+        "sigs.k8s.io/structured-merge-diff/v4"
+      ],
+      "github.com/modern-go/reflect2": [
+        "github.com/json-iterator/go",
+        "github.com/prometheus/client_golang",
+        "go.etcd.io/etcd/client/v2",
```
Comment (liggitt) [inline hack/unwanted-dependencies.json:221]: only via json-iterator, already dropped in https://github.com/etcd-io/etcd/pull/13439

File: hack/unwanted-dependencies.json [hunk 2]
```diff
@@ -204,6 +206,23 @@
...
+        "go.etcd.io/etcd/server/v3",
+        "go.opentelemetry.io/contrib/instrumentation/github.com/emicklei/go-restful/otelrestful",
+        "k8s.io/kube-openapi"
```
Comment (liggitt) [inline hack/unwanted-dependencies.json:224]: via structured-merge-diff → json-iterator, removal tracked in https://github.com/kubernetes-sigs/structured-merge-diff/issues/202

**Comment by aojea** [comment]
/lgtm

Thanks

**Pull request merged**